sdk/src/azure/core/az_hfsm.c

```diff
@@ -0,0 +1,164 @@
+/* Copyright (c) Microsoft Corporation. All rights reserved. */
+/* SPDX-License-Identifier: MIT */
+
+/**
+ * @file
+ * @brief Hierarchical Finite State Machine (HFSM) implementation.
+ *
+ * @details This implementation is _not_ providing complete HFSM functionality. The following
+ *          constraints must be made by the developer for their state machines:
+ *          1. A single top level state must exist.
+ *          2. Transitions can only be made to sub-states, peer-states and super-states.
+ *          3. The initial state is always the top-level state. Transitions must be made by the
+ *             application if an inner state must be reached during initialization.
+ */
+
+#include <stddef.h>
```

Comment:
Please standardize the order for include files (see az_core*.h/c files).
IIRC, first azure/ in lexicographical order then std / OS includes (also sorted).